test/primitives.jl

```diff
@@ -196,17 +196,18 @@ module TestPrimitives
 
     # for each, assert returned values are unwrapped and inplace change
     # dropnull!(X::NullableVector)
-    @test isequal(dropnull!(z), [1, 2, 3, 5, 7])
-    @test isequal(z, Nullable[1, 2, 3, 5, 7])
+    @test dropnull!(z) == [1, 2, 3, 5, 7]
+    @test isequal(z, NullableArray([1, 2, 3, 5, 7]))
 
     # dropnull!(X::AbstractVector)
-    @test isequal(dropnull!(A), [1, 2, 3, 5, 7])
+    @test dropnull!(A) == [1, 2, 3, 5, 7]
     @test isequal(A, Any[Nullable(1), Nullable(2), Nullable(3), Nullable(5),
                          Nullable(7)])
 
     # dropnull!(X::AbstractVector{<:Nullable})
-    @test isequal(dropnull!(B), [1, 2, 3, 5, 7])
-    @test isequal(B, Nullable[1, 2, 3, 5, 7])
+    @test dropnull!(B) == [1, 2, 3, 5, 7]
+    @test isequal(B, [Nullable(1), Nullable(2), Nullable(3), Nullable(5),
```

Comment:
That's much easier on the eyes, I like that. It loses the `{Int64}` information
```
julia> Nullable.([1, 2, 3, 5, 7])
5-element Array{Nullable,1}:
 1
 2
 3
 5
 7

julia> [Nullable(1), Nullable(2), Nullable(3), Nullable(5),
                                    Nullable(7)]
5-element Array{Nullable{Int64},1}:
 1
 2
 3
 5
 7
```

but I think losing that `{Int64}` is fine. The goal is to show that the functions handle `NullableArrays` as well as `Vectors{Nullable}`. I'll change the others as well, i.e.
```julia
[Nullable(1), Nullable(2), Nullable(3), Nullable(), Nullable(5), Nullable(), Nullable(7), Nullable()]
```
-> 
```julia
Nullable.([1, 2, 3, Nullable(), 5, Nullable(), 7, Nullable()])
```